boa/src/syntax/parser/expression/primary/object_initializer/mod.rs

```diff
@@ -141,10 +141,24 @@ where
             let node = AssignmentExpression::new(false, self.allow_yield, self.allow_await)
                 .parse(cursor)?;
             cursor.expect(Punctuator::CloseBracket, "expected token ']'")?;
-            cursor.expect(Punctuator::Colon, "expected token ':'")?;
-            let val = AssignmentExpression::new(false, self.allow_yield, self.allow_await)
-                .parse(cursor)?;
-            return Ok(node::PropertyDefinition::ComputedPropertyName(node, val));
+            let next_token = cursor.next()?.ok_or(ParseError::AbruptEnd)?;
+            match next_token.kind() {
+                TokenKind::Punctuator(Punctuator::Colon) => {
+                    let val = AssignmentExpression::new(false, self.allow_yield, self.allow_await)
+                        .parse(cursor)?;
+                    return Ok(node::PropertyDefinition::property(node, val));
+                }
+                TokenKind::Punctuator(Punctuator::OpenParen) => {
+                    return MethodDefinition::new(self.allow_yield, self.allow_await, node)
+                        .parse(cursor);
+                }
```

Comment:
I don't think this should be here. This allows parsing things like:
```Javascript
propobj = {
    name: "name",
    ["1" + 2 + 3](set sum(a, b) {a + b})
}
```
which is not valid Javascript.
Maybe you meant to put a check for this after `if cursor.next_if(Punctuator::OpenBracket)?.is_some()`?

Reply:
If I'm not completely mistaken, this should be right. `MethodDefinition` parses arguments and function body e.g. `(){}`.
This is for parsing things like:
```javascript
obj = {
    ["1" + 2](arg1) {
        console.log(arg1)
    }
}
```
Which is valid.

Your example actually fails like expected when I try it on this branch:
```
>> propobj = {
    name: "name",
    ["1" + 2 + 3](set sum(a, b) {a + b})
}
Uncaught: "SyntaxError": "expected token ',', got 'sum' in parameter list at line 3, col 23"
```

Reply:
Ah, I see. Maybe you could change `MethodDefinition` to `MethodSignatureBody`? Because the spec defines `MethodDefinition` as 
```
MethodDefinition[Yield, Await] :
    ClassElementName[?Yield, ?Await] ( UniqueFormalParameters[~Yield, ~Await] ) { FunctionBody[~Yield, ~Await] }
    GeneratorMethod[?Yield, ?Await]
    AsyncMethod[?Yield, ?Await]
    AsyncGeneratorMethod[?Yield, ?Await]
    get ClassElementName[?Yield, ?Await] ( ) { FunctionBody[~Yield, ~Await] }
    set ClassElementName[?Yield, ?Await] ( PropertySetParameterList ) { FunctionBody[~Yield, ~Await] }
```
or maybe could you change `MethodDefinition` to parse its own `PropertyName`?.

Reply:
Yeah I was a bit confused by this aswell. But I think we should not refacor that in this PR. There is already enough logic going on here. If you are ok with it, I will open an issue to refactor that parsing function.

Reply:
Yeah, that's ok 😃